Code/Source/Entity/ActorEntityManager.cpp

```diff
@@ -0,0 +1,157 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <EMotionFX/Source/ActorInstance.h>
+#include <EMotionFX/Source/Mesh.h>
+#include <EMotionFX/Source/Node.h>
+#include <EMotionFX/Source/SubMesh.h>
+#include <EMotionFX/Source/TransformData.h>
+#include <Entity/ActorEntityManager.h>
+#include <Utilities/RGLUtils.h>
+#include <rgl/api/core.h>
+
+namespace RGL
+{
+    ActorEntityManager::ActorEntityManager(AZ::EntityId entityId)
+        : EntityManager(entityId)
+    {
+        EMotionFX::Integration::ActorComponentNotificationBus::Handler::BusConnect(entityId);
+    }
+
+    ActorEntityManager::ActorEntityManager(ActorEntityManager&& other)
+        : EntityManager(std::move(other))
+    {
+        EMotionFX::Integration::ActorComponentNotificationBus::Handler::BusDisconnect();
+    }
+
+    ActorEntityManager::~ActorEntityManager()
+    {
+        for (MeshPair mesh : m_meshes)
+        {
+            RGL_CHECK(rgl_mesh_destroy(mesh.m_rglMesh));
+        }
+    }
+
+    void ActorEntityManager::Update()
+    {
+        UpdateMeshVertices();
+        EntityManager::Update();
+    }
+
+    void ActorEntityManager::OnActorInstanceCreated(EMotionFX::ActorInstance* actorInstance)
+    {
+        m_actorInstance = actorInstance;
+        EMotionFX::Actor* actor = actorInstance->GetActor();
+
+        static constexpr size_t LodLevel = 0UL; // Highest LOD.
+        const size_t NodeCount = actor->GetNumNodes();
+        for (size_t jointIndex = 0LU; jointIndex < NodeCount; ++jointIndex)
+        {
+            EMotionFX::Mesh* mesh = actor->GetMesh(LodLevel, jointIndex);
+            rgl_mesh_t rglMesh;
+            if (mesh && (rglMesh = EMotionFXMeshToRglMesh(*mesh)))
+            {
+                m_meshes.push_back({ mesh, rglMesh });
+            }
+        }
+
+        m_entities.reserve(m_meshes.size());
+        for (MeshPair& mesh : m_meshes)
+        {
+            rgl_entity_t entity = nullptr;
+            Utils::SafeRglEntityCreate(entity, mesh.m_rglMesh);
+            if (entity)
+            {
+                m_entities.emplace_back(entity);
+            }
+        }
+
+        if (!m_entities.empty())
+        {
+            UpdatePose();
+        }
+    }
+
+    void ActorEntityManager::UpdateMeshVertices()
+    {
+        m_actorInstance->UpdateMeshDeformers(0.0f);
+
+        for (MeshPair& mesh : m_meshes)
+        {
+            UpdateVertexPositions(*mesh.m_eMotionMesh);
+            RGL_CHECK(rgl_mesh_update_vertices(mesh.m_rglMesh, m_positions.data(), m_positions.size()));
```

Comment:
There is no need to have `m_positions` member variable.
`UpdateVertexPositions` should be renamed to `GetVertexPositions` (or `GetMeshVertices`) and return the vector of vertices.